timemachine/cpp/src/kernels/k_fixed_point.cuh

```diff
@@ -67,3 +67,30 @@ template <typename RealType> unsigned long long __device__ __forceinline__ FLOAT
 template <typename RealType> unsigned long long __device__ __forceinline__ FLOAT_TO_FIXED_BONDED(RealType v) {
     return static_cast<unsigned long long>(real_to_int64(v * FIXED_EXPONENT));
 }
+
+/* FLOAT_TO_FIXED_ENERGY converts floating point energies into fixed point. Values beyond LLONG_MAX/LLONG_MIN
+* (or nonfinite) will be capped to LLONG_MAX. This is accumulated into __int128 which handles the over and underflows,
```

Comment:
nit: does this detect underflow?

(my understanding is that it detects positive or negative overflows)

Reply:
Will fix to say 'negative overflow'

Reply:
Fixed in [6da81a0](https://github.com/proteneer/timemachine/pull/1090/commits/6da81a094c6812b11b64a39b78e8e3726948074e)